docs/source/design/dataset_argument.rst

```diff
@@ -0,0 +1,62 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
+
+********************
+``dataset`` argument
+********************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+All commands which operate on datasets have a ``dataset`` argument (``-d`` or
+``--dataset`` for the command line API) to identify a single dataset as the
```

Comment:
```suggestion
``--dataset`` for the :term:`CLI`) to identify a single dataset as the
```
and add CLI to glossary? After all Command Line Interface is not necessarily used as an API but rather as an "interactive" interface.